src/state_history_kv_tables.hpp

```diff
@@ -69,157 +70,172 @@ EOSIO_REFLECT(block_info_v0, num, id, timestamp, producer, confirmed, previous,
 using block_info = std::variant<block_info_v0>;
 
 // todo: move out of "state"?
-struct block_info_kv : eosio::table<block_info> {
-   index primary_index{ eosio::name{ "primary" }, [](const auto& var) {
-                          return std::visit(
-                                [](const auto& obj) { return eosio::check(eosio::convert_to_key(obj.num)).value(); },
-                                var);
-                       } };
-
-   index id_index{ eosio::name{ "id" }, [](const auto& var) {
-                     return std::visit(
-                           [](const auto& obj) { return eosio::check(eosio::convert_to_key(obj.id)).value(); }, var);
-                  } };
-
-   block_info_kv(eosio::kv_environment environment) : eosio::table<block_info>{ std::move(environment) } {
+struct block_info_kv : eosio::kv_table<block_info> {
+   index<std::vector<char>> primary_index{
```

Comment:
The temporary wrapper expected keys to already be serialized. It'd be nice to switch indexes to return real types since `kv_table` supports that.